### ISSUE TITLE
PHPUnit 10 deprecation on non-static data provider in AnalysisTrait

### ISSUE THREAD

**Issue by jnoordsij** [GrahamCampbell/Analyzer#10]
While running tests for https://github.com/GrahamCampbell/Laravel-TestBench/pull/31, I noticed a deprecation warning from PHPUnit:
```
There was 1 PHPUnit deprecation:

1) GrahamCampbell\Tests\TestBench\AnalysisTest::testReferences
Data Provider method GrahamCampbell\Tests\TestBench\AnalysisTest::provideFilesToCheck() is not static
```

To be compatible with future PHPUnit versions, this should probably be resolved.

**Comment by GrahamCampbell** [comment]
We don't support PHPUnit 10 at this time.

**Comment by GrahamCampbell** [comment]
I've released v4.0.0, which supports PHPUnit 9 and 10, only. Note that there are breaking changes that will require changes when upgrading. See https://github.com/GrahamCampbell/Analyzer/releases/tag/v4.0.0.
